src/canvas/helper.ts

```diff
@@ -4,6 +4,23 @@ import { GradientObject } from '../graphic/Gradient';
 import { RectLike } from '../core/BoundingRect';
 import Path from '../graphic/Path';
 
+export function isSafeNum(num: number, nonNegative = false) {
+    // null
+    if (num === null) {
+        return false;
+    }
+    // NaN, Infinity, undefined, 'xx'
+    if (!isFinite(num)) {
```

Comment:
`!isFinite` also includes `num === null` check.

These three check can be merged to:

```ts
if (!isFinite(num) || (nonNegative && num < 0 {
  ...
}
```

Reply:
<img width="174" alt="image" src="https://user-images.githubusercontent.com/8437666/159125089-4235d15f-dd4c-4180-8dde-dd5e73e72f05.png">

Reply:
Sorry it's my mistake. `null` can be convert to `0` here. I rechecked if `null` can be used as argument of `createLinearGradient` and seems it can. But `undefined` will throw an `non-finite` exception. So if we mainly wan using this check to avoid exceptions, we can still remove `null` check.